docs/api/authentication/api-keys.md

```diff
@@ -7,72 +7,69 @@ description: Learn what Momento API keys are, how to create them, and how they a
+## Roles
 
-### Step 1: Sign up or log into the Momento console
+API keys support roles so you can control who can manage or use resources:
 
-Go to the [Momento console](https://console.gomomento.com/tokens) and follow the instructions to log in with your email address, Google account, or GitHub account.
+- **Owner**: full access to all resources, including API keys and account membership.
+- **Operator**: full access to API/data resources; read-only for account membership.
+- **Viewer**: read-only access to all resources.
 
-![Image of Momento console landing page](@site/static/img/getting-started/console.png)
+## Create an API key
 
-### Step 2: Generate your API key
+Create keys in the [Momento console](https://console.gomomento.com/api-keys) or with the Auth APIs. Choose a role, add an optional description, and optionally set an expiry. The full key string is shown once; copy it and store it securely in a secret manager such as AWS Secrets Manager, Azure Key Vault, or GCP Secret Manager.
 
-In the console, select the [API Keys](https://console.gomomento.com/tokens) menu option.
+## Manage API keys
 
-Once on the API key page, select the information that matches where your caches live:
+### List
+List active keys to view metadata (key ID, role, issuedAt, optional expiresAt, description). The plaintext key is never returned.
 
-1. Cloud provider
-2. Region
-3. Key Type
-3. (Optional) Expiration date
+### Revoke
+Revoke a key to disable it immediately. If you set an expiry, rotate before it lapses to avoid outages.
 
-![Image showing the fields to create a new API key](@site/static/img/getting-started/select-provider-region.png)
+## Migrate from legacy API keys
 
-Once complete, click on the **Generate** button to create your API Key!
-
-You can copy the value of the key directly and store it in a secure location or you can click on the **Download JSON** button to download the key and expiration date to your machine.
-
-### Step 3: Secure it!
-
-API keys are long-lived and typically grant high levels of access to the holder. With this in mind, be sure to securely store your API key in a location that encrypts the value and prevents plain-text viewing.
-
-If your application is hosted in the cloud, it is best practice to store your keys in and retrieve from services like [AWS Secrets Manager](https://aws.amazon.com/secrets-manager/), [Azure Key Vault](https://learn.microsoft.com/en-us/azure/key-vault/general/overview), or [GCP Secret Manager](https://cloud.google.com/secret-manager).
+:::info
+Legacy API keys will continue working, but next time you rotate your keys, you'll be automatically upgraded to the new API key format. These keys require an endpoint to be provided in the credential provider, **resulting in a breaking change**. Below is the migration path from the old credential provider to the new provider.
```

Comment:
I think you mean "Migrating from v1 API keys"? "Legacy api keys" is another type of api key that predates v1 keys.

Unless you're using the word "Legacy" here to mean anything older than v2 keys.


> but next time you rotate your keys, you'll be automatically upgraded to the new API key format

I don't think you get a new v2 key by "rotating" v1/legacy keys? Do you mean like next time customer goes into the console to generate a key they'll get a v2 key?